File: go/internal/models/destinationModel.go
package models

type Destination struct {
	//gorm.Model
	ID         uint       `gorm:"primary_key; not null" json:"id"`
	Name       string     `gorm:"size: 255;not null" json:"name"`
	Image      string     `gorm:"size: 255;null" json:"image"`
	Activities []Activity `gorm:"many2many:destination_activities; null" json:"activities"`
	Hotels     []Hotel    `gorm:"many2many:destination_hotels; null" json:"hotels"`
}

File: flutter_app/lib/models/destination_model.dart
import 'package:flutter_app/models/hotel_model.dart';

import 'activity_model.dart';

class Destination {
  int id;
  String image;
  String name;
  List<Activity> activities;
  List<Hotel> hotels;
  Destination({
    required this.id,
    required this.image,
    required this.name,
    required this.activities,
    required this.hotels,
  });

  Destination.fromJson(Map<String, dynamic> json)
      : id = json['id'],
        image = json['image'],
        name = json['name'],
        activities = (json['activities'] as List)
            .map((activityJson) => Activity.fromJson(activityJson))
            .toList(),
        hotels = (json['hotels'] as List)
            .map((hotelJson) => Hotel.fromJson(hotelJson))
            .toList();
}
